src/utils/runtime-context.ts

```diff
@@ -49,11 +51,22 @@ export class RuntimeContext {
     }
   }
 
+  triggerUserChange() {
+    globals.refetchUser();
+    this.runCallbacks();
+  }
+
+  onUserChanged(callback: () => void) {
+    this._userChangedListeners.add(callback);
+  }
```

Comment:
I know this is not perfect, maybe later we can have more proper event-driven/reactive approach. Other things works nicely with VSC build in APIs (like observing settings, or fs watching), os it depends what we will need further the road.